pip/index.py

```diff
@@ -412,10 +413,17 @@ def _find_all_versions(self, project_name):
         for location in url_locations:
             logger.debug('* %s', location)
 
+        formats = set(["source"])
+        if self.use_wheel:
+            formats.add("binary")
+        search = Search(
+            project_name.lower(),
+            pkg_resources.safe_name(project_name.lower()).lower(),
```

Comment:
This is actually deliberate: once you trace the prior code through, this is what was happening. I figured preserving it would at minimum provoke discussion :). I can obviously remove one of the lowers if you'd like!